### PR TITLE
Fix default detection

### LINKED ISSUE #3153: Interactive webroot values not stored in renewal config file
This is due to using mutable values as the default in `argparse` such as `list` and `dict` and then reaching inside `argparse` to check the defaults in `storage.py`. The result is modifications to these values also modify the default, meaning they always compare equal to the default value and `storage.py` thinks they should not be saved.

### PR DESCRIPTION
**Pull request by bmw**
Fixes #3153.

### CODE REVIEW

**Comment by pde** [comment]
Should `set_by_cli` return true for things that are actually set interactively? It seems that it possibly should (and should perhaps be renamed accordingly?).

**Comment by bmw** [comment]
So I could move the logic that I added to `storage.py` to `set_by_cli`, but something like the `DEFAULTS` mechanism is still necessary since the dummy arguments used by `set_by_cli` aren't passed through the entire program.

**Comment by pde** [comment]
I understand that we'll still need `DEFAULTS`, it just seems that we should aim to have one code path for detecting whether variables have their default values, rather than (if I'm reading it correctly) two variants of this logic.

File: certbot/storage.py [hunk 1]
```diff
@@ -162,30 +162,16 @@ def relevant_values(all_values):
...
+        # (it is set_by_cli(), we don't know the default value, or
+        # the current value differs from the default value).
         if _relevant(option):
-            if (cli.set_by_cli(option)
-                or not _is_cli_default(option, value)):
-#                or option not in constants.CLI_DEFAULTS
-#                or constants.CLI_DEFAULTS[option] != value):
+            if (cli.set_by_cli(option) or
+                    option not in cli.DEFAULTS or
+                    cli.DEFAULTS[option] != value):
```
Comment (pde) [inline certbot/storage.py]: I think it would be fine to wrap these two boolean tests in a simple `cli.is_default` function and merge.

Comment (bmw) [inline certbot/storage.py]: I may have misunderstood you. I thought you wanted a common function in `cli.py`. Either way, I'm almost done.

File: certbot/tests/storage_test.py [hunk 2]
```diff
@@ -533,10 +533,9 @@ def test_relevant_values_default(self, mock_parser):
         """Test that relevant_values() can reject a default value."""
         # pylint: disable=protected-access
         from certbot import storage
-        mock_parser.verb = "certonly"
         mock_parser.args = ["--standalone"]
-        mock_action = mock.Mock(dest="rsa_key_size", default=2048)
-        mock_parser.parser._actions = [mock_action]
+        mock_parser.defaults = {"rsa_key_size": 2048}
```
Comment (TheNavigat) [inline certbot/tests/storage_test.py]: What about this? https://github.com/certbot/certbot/blob/f9d5ecaf6fc6776988ddf5ea9dbb010ad45eeb7f/certbot/tests/storage_test.py#L548

Comment (bmw) [inline certbot/tests/storage_test.py]: Nice catch.

I rewrote all `relevant_values` test in my latest commit.

**Comment by TheNavigat** [comment]
LGTM

**Comment by pde** [comment]
LGTM

**Pull request merged**